ruby_basics/11_nested_collections/exercises/nested_hash_exercises.rb

```diff
@@ -0,0 +1,32 @@
+def add_information_about_language(languages, language_name, info_key, info_value)
+  # Take languages (a hash with language names as keys and information about
+  # each language in a nested hash as the corresponding value) and add the
+  # key/value pair info_key/info_value to the nested hash of language_name,
+  # then return the updated languages hash
+end
+
+def add_language(languages, language_name, language_info_value)
+  # Take languages and add the key/value pair language_name/language_info_value
+  # to it, then return languages
+end
+
+def delete_information_about_language(languages, language_name, info_key)
+  # Take languages and delete the key/value pair with key info_key from
+  # language_name, then return languages
+end
+
+def delete_language(languages, language_name)
+  # Take languages and delete the language_name key/value pair, then return
+  # languages
+end
+
+def delete_beautiful_languages(languages)
+  # Take languages and delete any language that has the key/value pair
+  # { is_beautiful => true } listed in its information, then return languages
+end
```

Comment:
I think `is_beautiful` needs a `?`. Also, this test seems to be the only one where it is okay to mutate the provided argument `languages`. I just wanted to make sure that was the intention here, as opposed to returning a new nested hash without the beautiful languages.

Reply:
Agree on `is_beautiful`.

Hmmm I think most of them can be done with mutation of the provided argument, and I specified when it couldn't. I will have to double-check this though.

Reply:
@rlmoser99 unless I'm not seeing something right the only exercise where mutation isn't allowed is count_empty_seats, but I could be blind to something here so please correct me if I'm wrong.

Reply:
@wise-king-sullyman You are right... plus, this one comes right after 2 other deleting methods!

However, I think what I was really thinking is that it is better practice to return a new item than to mutate the original item, or at least that is the habit that I've formed. I solved this one using `delete_if` at first, but then changed to `#select` because that is the way that I would do it in my own code. Which then probably made me feel like the name of the method & the instructions were not quite lining up like they should.

My preference would be to have students `find_beautiful_languages` rather than to delete them, but I think I am going to need @CouchofTomato to weigh in with his thoughts on this one. I don't think we have to always follow best practices, especially if we think students need practice using a method like `delete_if` (which is also not covered in our curriculum and may need a hint to look in the docs).

Reply:
`is_beautiful` issue resolved in 83cf5a4

You're right, the best practice would be to not mutate unless needed. Though there are a lot of other exercises here that may need to be changed as well if that is the standard being set. Fine with doing so of course.

Alternatively, if you do just think this one exercise in particular needs to change I will update it by itself.

Reply:
I think the other exercises that mutate the array are for the students to practice using a method they learned about in the lesson. This example is for students to practice iterating through the hash, so I prefer for them to approach it using `#select`. They could still solve it using `#delete_if`, but since this method is not covered in the lesson it would need another hint to look in the docs (again).

Reply:
Resolved in 59bcedf90c6448dbbb5edf106df83abeed453008